core/frontend/src/DisplayStyleState.ts

```diff
@@ -422,16 +432,30 @@ export abstract class DisplayStyleState extends ElementState implements DisplayS
     return this.settings.mapImagery.baseTransparency;
   }
 
-  /** @internal  */
+  /** @beta  */
   public changeBaseMapTransparency(transparency: number) {
     if (this.settings.mapImagery.backgroundBase instanceof ColorDef) {
       this.settings.mapImagery.backgroundBase = this.settings.mapImagery.backgroundBase.withTransparency(transparency * 255);
       this._synchBackgroundMapImagery();
+    } else if (this.settings.mapImagery.backgroundBase instanceof BackgroundMapProvider) {
+      this.settings.mapImagery.backgroundBase = this.settings.mapImagery.backgroundBase.clone({providerData: {transparency}});
+      this._synchBackgroundMapImagery();
     } else {
       this.changeBaseMapProps({ transparency });
     }
   }
 
+  /** @beta  */
+  public changeBaseMapVisible(visible: boolean) {
+    // TODO: There is no way actually to change the visibility of the base map when its sets to a ColorDef.
```

Comment:
Seems like API shouldn't allow this then.

Reply:
Are you suggestion that `changeBaseMapVisible ` should check if `backgroundBase` is a ColorDef, and return an error/throw?

Reply:
Ideally we don't provide users with APIs that can't perform their advertised function in the first place.